### DISCUSSION TITLE
Query Regarding Lack of Simultaneous RGBD and Infrared (Infra1, Infra2) Data Streaming Feature

### DISCUSSION THREAD

**Discussion by wataru-okumura** [realsenseai/realsense-ros#3095] [Q&A]
To whom it may concern,

I am writing to inquire about the absence of the feature enabling simultaneous delivery of RGBD and infrared (infra1, infra2) data (an extension of the RGBD.msg). As I understand that this functionality is technically feasible, I would appreciate it if you could provide the reasons for its non-provision.

Specifically, I am interested in whether this is due to technical constraints or other reasons. Additionally, I would be grateful if you could share any future prospects for this feature and any plans under consideration.

I understand that you are very busy, but I would greatly appreciate your guidance on this matter.

Sincerely,

**Comment by MartyG-RealSense** [comment]
Hi @wataru-okumura  There is no problem with streaming RGB color from the RGB sensor (if the camera has one) and infrared from both the left and right infrared sensors (Infra1 and Infra2).  However, Infra2 is only available on a USB 3 connection.  On USB 2.1 only Infra1 is supported.

All RealSense 400 Series USB camera models except D405 have an RGB sensor (D405 provides RGB from its depth sensor as it does not have a dedicated RGB sensor component).  

With the **depth module** circuit board versions of the cameras, all can provide Infra1 and Infra2 (if used with USB 3) but not all of the modules are equipped with an RGB sensor.

**Reply by wataru-okumura** [reply]
Thanks for the answers.
Glad to know who can retrieve each stream.
I am asking if there is a way to get RGBD and IR images at the same time, which is a feature this repository offers.
I know that by subscribing to RGBD.msg, I can get the color and depth images at the same time, but in addition to that, I would like to get the data that is being delivered at the same time for the infrared images as well. I don't see any ROS topic for that, but is it possible to do that either by changing the settings or by user customization of the code?

**Comment by MartyG-RealSense** [comment]
Instead of using RGBD.msg, I would recommend enabling depth, color, infra1 and infra2 and using the **enable_sync** parameter to gather the closest frames of different sensors (infrared, color and depth) to be sent with the same timetag.

`ros2 launch realsense2_camera rs_launch.py enable_depth:=true enable_color:=true enable_infra1:=true enable_infra2:=true enable_sync:=true`

**Comment by wataru-okumura** [comment]
Sorry for the late reply.
I have successfully confirmed that I can subscribe to multiple streams of images with the same timestamp!

**Comment by MartyG-RealSense** [comment]
It's no problem at all.  I'm pleased to hear that you succeeded.  Thanks very much for the update!
